src/firmware/app/control/trajectory_planner.h

```diff
@@ -72,232 +71,369 @@ typedef enum TrajectoryPlannerGenerationStatus
- * copied into
- * @param max_allowable_speed_profile [in] The pre-allocated array that will contain all
- * of the maximum allowable velocity values for each point on the curve
- * @param arc_segment_length [in] The length of each arc length segment in meters
- * @param max_allowable_acceleration [in] The max allowable acceleration at any point in
- * m/s^2
- * @param max_allowable_speed [in] The max allowable speed at any point in m/s
- *
- * */
-void app_trajectory_planner_generateForwardsContinuousVelocityProfile(
-    const unsigned int num_segments,
-    float velocity_profile[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS],
-    float max_allowable_speed_profile[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS],
-    const float arc_segment_length, const float max_allowable_acceleration,
-    const float max_allowable_speed);
+ * Function generates the time profile for a speed profile with given segment lengths and
+ * start and end velocities. This is done using a constant acceleration assumption over
+ * the total displacement of each segment
+ *
+ * @pre All arrays must be pre-allocated up to at least
+ * TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS
+ *
+ * @param segment_lengths [in] The length of each segment in meters. The segment length is
+ * the distance travelled between successive speed elements.
+ *
+ * @param speeds [in] The speed profile defining the current speed at each element. In
+ * m/s.
+ *
+ * @param trajectory_duration [in] The duration between successive speed elements in the
+ * profile. In seconds.
+ *
+ * @param num_elements [in] The number of elements in the speed profile.
+ */
+void app_trajectory_planner_generatePositionTrajectoryTimeProfile_2(
+    float segment_lengths[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS],
+    float speeds[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS],
+    float trajectory_duration[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS],
+    unsigned int num_elements);
 
 /**
- * Modifies a forwards continuous velocity profile to also be backwards continuous based
- * on the input parameters.
+ * Function balances the segment time durations of 2 separate speed profiles. The segment
+ * durations of profile1 and profile2 will be made equal - to the LONGEST duration at each
+ * point along the profiles. This is done by modifying the final speed elements.
  *
- * NOTE: This function does not break forwards continuity of the trajectory
+ * @pre The profile must have the same number of elements. Other than the displacement
+ * arrays that must have 1 fewer element.
  *
- * @pre The input velocity_profile is required to be FORWARDS CONTINUOUS before it is used
- * as an input to this function
- * @pre forwards_continuous_velocity_profile is pre-allocated up to at least
- * TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS
+ * @pre The arrays must be pre-allocated up to at least
+ * TRAJECTORY_PLANNER_MAX_NUM_SEGMENTS.
+ *
+ * @param speeds1 [in] The speed profile corresponding to trajectory 1. In m/s.
```

Comment:
Disclaimer: so far only read the header file

IIRC the use for this function was to balance the linear and angular profiles so that the orientation is tracked as expected over the duration of the path. Is that correct? Currently the javadoc references everything in m/s and makes it sound like it operates only on linear profiles.

Reply:
Specified that the denominator must be seconds and removed the numerator unit, whether it's rad/meters